backend/src/v5/services/journaling/index.js

```diff
@@ -0,0 +1,120 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { logInvitationAction, logPermissionAction, logUserAction } = require('../../models/teamspaces.audits');
+const Yup = require('yup');
+const { actions } = require('../../models/teamspaces.audits.constants');
+const { events } = require('../eventsManager/eventsManager.constants');
+const { logger } = require('../../utils/logger');
+const { subscribe } = require('../eventsManager/eventsManager');
+const { types } = require('../../utils/helper/yup');
+
+const userAdded = async ({ teamspace, executor, user }) => {
+	try {
+		await logUserAction(teamspace, actions.USER_ADDED, executor, user);
+	} catch (err) {
+		logger.logError(`Failed to add a ${actions.USER_ADDED} audit log`);
+	}
+};
+
+const userRemoved = async ({ teamspace, executor, user }) => {
+	try {
+		await logUserAction(teamspace, actions.USER_REMOVED, executor, user);
+	} catch (err) {
+		logger.logError(`Failed to add a ${actions.USER_REMOVED} audit log`);
+	}
+};
+
+const permissionsSchema = Yup.array().of(Yup.string()).nullable().min(1)
```

Comment:
there's no point having default(null) if you never user the return value from validate.

It'll never get to assign the default